agent/internal/grpc/grpc.go

```diff
@@ -43,13 +43,19 @@ type GrpcConnectionParams struct {
 }
 
 type DeployFunc func(context.Context, *dogger.DeploymentLogger, *v1.DeployImageRequest, *v1.VersionData) error
-type WatchFunc func(string) []*crux.ContainerStatusItem
+type WatchFunc func(context.Context, string) []*crux.ContainerStatusItem
 
 type WorkerFunctions struct {
 	Deploy DeployFunc
 	Watch  WatchFunc
 }
 
+type contextKey int
+
+const (
+	contextConfigKey contextKey = iota
```

Comment:
IMHO if there is no more value, you don't need to add to a multiple const form.
 
```
const contextConfigKey contextKey = iota
```

What is your opinion? @nandor-magyar

Reply:
iota is generally used in enums, what its purpose here?
If it is constant it could more more verbose, if an enum with a single entry it could more explicit.

Reply:
Currently it's just a const, in the future if we'd like to add anything else to the context, I think this leaves open the possibility.